backend/stakepoold/rpc/rpcserver/client.go

```diff
@@ -0,0 +1,168 @@
+package rpcserver
+
+import (
+	"sync"
+	"time"
+
+	"github.com/decred/dcrd/rpcclient/v3"
+)
+
+const (
+	// connectionRetryInterval is the amount of time to wait in between
+	// retries when automatically reconnecting to an RPC server.
+	connectionRetryInterval = time.Second * 5
+	// disconnectCheckInterval is the amount of time to wait between
+	// checks for a disconnection.
+	disconnectCheckInterval = time.Second * 10
+)
+
+// Client holds the information related to an rpcclient and handles access to
+// that client through a mutex.
+//
+// It should be noted that this is a temporary fix to the problem that rpcclient
+// does not return an error when autoreconnect is turned on but the client is
+// disconnected. The permanent solution is to change the behaviour of rpccleint.
+// TODO: Remove this file.
+type Client struct {
+	// client is protected by a mutex that must be held for reads/writes.
+	client *rpcclient.Client
+	mux    sync.RWMutex
+
+	cfg          *rpcclient.ConnConfig
+	ntfnHandlers *rpcclient.NotificationHandlers
+	wg           sync.WaitGroup
+	stop         chan struct{}
+
+	connected    chan struct{}
+	connectedMux sync.Mutex
+}
+
+// Connected returns a receiving copy of the current connected channel. If
+// disconnected and the channel is not yet blocking, creates a new channel that
+// will be closed on a successful reconnect.
+func (c *Client) Connected() <-chan struct{} {
+	c.connectedMux.Lock()
+	defer c.connectedMux.Unlock()
+	if c.RPCClient().Disconnected() {
+		// Start blocking on connected chan if not already.
+		select {
+		case <-c.connected:
+			c.connected = make(chan struct{})
```

Comment:
Is this expected to every be the case? (A disconnected `rpcclient.Client`, but `c.connected` still closed indicating that `Client` thinks it is connected?

Reply:
Yes.
If the connected channel was not opened by Connected(), it will be closed here. Which will be most of the time.